inc/utils/content.php

```diff
@@ -578,32 +577,41 @@ function odm_excerpt($the_post, $num = 45, $read_more = '')
 
 		$stripped_content = strip_tags($translated_content);
 		$stripped_content = strip_shortcodes($stripped_content);
-		if(trim($stripped_content)):
-			$stripped_content = preg_replace( "/\&hellip;/",'', trim($stripped_content));
-			$stripped_content_arr = explode(' ', trim($stripped_content), $num+1);
-
-			if($stripped_content_arr):
-				array_splice($stripped_content_arr, $limit);
-				$excerpt_content = implode(' ', $stripped_content_arr);
-		    if (odm_language_manager()->get_current_language() == "km"):
-		  		$excerpt_zeo_space = explode("&#8203;", $excerpt_content, $num+1); //explode by zerowidthspace​
-					array_splice($excerpt_zeo_space, $limit);
-		  		$excerpt_content = implode("&#8203;", $excerpt_zeo_space); //implode by zerowidthspace
-				endif;
 
-				$color_name = odm_country_manager()->get_current_country().'-color';
-				if ($read_more != ''):
-					$excerpt_words =  $excerpt_content." ... <a href='".get_permalink($post->ID)." ' class='".$color_name."'>".__($read_more, 'odm').'</a>';
-				else:
-					$excerpt_words = $excerpt_content." <a href='".get_permalink($post->ID)." ' class='".$color_name."'>...</a>";
-				endif;
-				return '<p>' . $excerpt_words . '</p>';
-			endif;
+		$stripped_content = preg_replace( "/\&hellip;/",'', trim($stripped_content));
+
+		$excerpt_content = shorten_string_words($stripped_content,$limit,odm_language_manager()->get_current_language());
 
-			return '<p>' . $stripped_content . '</p>';
+		$color_name = odm_country_manager()->get_current_country().'-color';
+		if ($read_more != ''):
+			$excerpt_words =  $excerpt_content." ... <a href='".get_permalink($post->ID)." ' class='".$color_name."'>".__($read_more, 'odm').'</a>';
+		else:
+			$excerpt_words = $excerpt_content." <a href='".get_permalink($post->ID)." ' class='".$color_name."'>...</a>";
 		endif;
+		return '<p>' . $excerpt_words . '</p>';
+
  }
 
+function shorten_string_words($string, $limit = 40, $current_lang = "en"){
```

Comment:
> Will the new function be call to use in some where?

Yes, so far in 2 places as described on https://github.com/OpenDevelopmentMekong/wp-odm_theme/issues/1167 but some more to come I would expect.